cwms-data-api/src/main/java/cwms/cda/data/dto/forecast/ForecastInstance.java

```diff
@@ -1,130 +1,283 @@
-  private Instant firstDateTime;
-
-  @XmlAttribute(name = "last-date-time")
-  private Instant lastDateTime;
-
-  @XmlAttribute(name = "max-age")
-  private Integer maxAge;
-
-  @XmlAttribute(name = "time-series-count")
-  private Integer timeSeriesCount;
-
-  @Schema(description = "Forecast Instance Notes")
-  @XmlAttribute
-  private String notes;
-
-  @XmlAttribute(name = "metadata")
-  private Map<String, String> metadata;
-
-  @Schema(description = "Forecast Filename")
-  @XmlAttribute(name = "filename")
-  private String filename;
-
-  @Schema(description = "Description of Forecast File")
-  @XmlAttribute
-  private String fileDescription;
-
-  @Schema(description = "Forecast File binary data")
-  @XmlElement(name = "forecast-data")
-  private byte[] fileData;
-
-  @SuppressWarnings("unused") // required so JAXB can initialize and marshal
-  private ForecastInstance() {}
-
-  public ForecastInstance(String specId, Instant dateTime, Instant issueDateTime,
-                          Instant firstDateTime, Instant lastDateTime, Integer maxAge, Integer timeSeriesCount,
-                          String notes, Map<String, String> metadata, String filename, String fileDescription,
-                          byte[] fileData) {
-    this.specId = specId;
-    this.dateTime = dateTime;
-    this.issueDateTime = issueDateTime;
-    this.firstDateTime = firstDateTime;
-    this.lastDateTime = lastDateTime;
-    this.maxAge = maxAge;
-    this.timeSeriesCount = timeSeriesCount;
-    this.notes = notes;
-    this.metadata = metadata;
-    this.filename = filename;
-    this.fileDescription = fileDescription;
-    this.fileData = fileData;
-  }
-
-  public String getSpecId() {
-    return specId;
-  }
-
-  public Instant getDateTime() {
-    return dateTime;
-  }
-
-  public Instant getIssueDateTime() {
-    return issueDateTime;
-  }
-
-  public Instant getFirstDateTime() {
-    return firstDateTime;
-  }
-
-  public Instant getLastDateTime() {
-    return lastDateTime;
-  }
-
-  public Integer getMaxAge() {
-    return maxAge;
-  }
-
-  public Integer getTimeSeriesCount() {
-    return timeSeriesCount;
-  }
-
-  public String getNotes() {
-    return notes;
-  }
-
-  public Map<String, String> getMetadata() {
-    return metadata;
-  }
-
-  public String getFilename() {
-    return filename;
-  }
-
-  public String getFileDescription() {
-    return fileDescription;
-  }
-
-  public byte[] getFileData() {
-    return fileData;
-  }
-
-  public void validate() throws FieldException {
-    //TODO
-  }
+@FormattableWith(contentType = Formats.JSONV2, formatter = JsonV2.class)
+@JsonDeserialize(builder = ForecastInstance.Builder.class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonNaming(PropertyNamingStrategies.KebabCaseStrategy.class)
+public class ForecastInstance extends CwmsDTO {
+
+    // unique identifier: spec + issueDateTime
+    // to find spec we need officeId, specId and locationId
+    // we get officeId from CwmsDTO

```

Comment:
I'm not sure if this class should have officeId, specId, locationId   
or should it just have the entire ForecastSpec?
or does it actually not need any of those in the actual object and the Controller methods (mostly create and update) should just take extra parameters?

Reply:
office, spec id, and designator would be the unique identifiers. Still not sure if the location is included in that

Reply:
confirmed, location id is not included in the unique constraint